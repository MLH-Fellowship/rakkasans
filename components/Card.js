import React from "react";
import { View, StyleSheet } from "react-native";

const Card = (props) => {
  return (
    <View style={{ ...styles.card, ...props.style }}>{props.children}</View>
  );
};

const styles = StyleSheet.create({
  card: {
    elevation: 5, // android
    shadowColor: "black", // shadow - ios
    shadowRadius: 8,
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.26,
    backgroundColor: "transparent",
    padding: 15,
    borderRadius: 15,
  },
});

export default Card;
