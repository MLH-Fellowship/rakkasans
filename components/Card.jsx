import React from 'react';
import { View, StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  card: {
    elevation: 5, // android
    shadowColor: 'black', // shadow - ios
    shadowRadius: 8,
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.26,
    backgroundColor: 'transparent',
    padding: 15,
    borderRadius: 15,
  },
});

const Card = (props) => {
  const { style, children } = props;
  return (
    <View style={{ ...styles.card, ...style }}>{children}</View>
  );
};

export default Card;
