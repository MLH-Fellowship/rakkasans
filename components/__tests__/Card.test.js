import React from 'react';
import renderer from 'react-test-renderer';

import Card from '../Card';

jest.useFakeTimers();

it('renders correctly', () => {
  const tree = renderer.create(<Card />).toJSON();
  expect(tree).toMatchSnapshot();
});
