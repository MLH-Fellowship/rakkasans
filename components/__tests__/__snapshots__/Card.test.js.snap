// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "transparent",
      "borderRadius": 15,
      "elevation": 5,
      "padding": 15,
      "shadowColor": "black",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.26,
      "shadowRadius": 8,
    }
  }
/>
`;
